src/main/java/org/truffleruby/core/symbol/SymbolNodes.java

```diff
@@ -44,6 +44,16 @@
 @CoreModule(value = "Symbol", isClass = true)
 public abstract class SymbolNodes {
 
+    @CoreMethod(names = "all_symbols", onSingleton = true)
+    public abstract static class AllSymbolsNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        protected DynamicObject allSymbols() {
+            return createArray(getContext().getSymbolTable().allSymbols());
```

Comment:
Our Array strategies don't support `RubySymbol[]`, only `Object[]`.

Reply:
I'll add an `assert` in `createArray()` to catch this (unless you do it already)

Reply:
Fixed.